torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -141,6 +178,9 @@ def decompress(fut):
         q = fut.value()[0].div_(world_size)
         torch.matmul(p, q.t(), out=matrix)
 
+        if state.use_error_feedback:
+            # Memorize the local errors.
+            state.error_dict[input_tensor] = input_tensor_cp - input_tensor
```

Comment:
As I understand this algorithm, iteration i + 1 needs iteration i's `input_tensor`, to save memory usage, can we clean up the unneeded tensors from the dict from previous states and would this improve the memory consumption situation?

Reply:
The error tensor stored in the state will always be updated in place, and there is only a single state instance across iterations.

`input_tensor_cp` is a temporary variable, and the memory will be released automatically after running this hook.

Reply:
You're right, the update is in place. Thanks for the catch!